### ISSUE TITLE
[web] - Bad state: No element -> 2

### ISSUE THREAD

**Issue by Macacoazul01** [bosskmk/pluto_grid#380]
flutter version: `2.10.0 `
package version: `pluto_grid: ^2.9.2`

I don't know if it's related to the same error of #331 (as the 2.9.0 update should have fixed that), but i'm having a similar problem:

The usage of a controller seems to break something in the state management of pluto. Everything renders fine but there's an error on the debugger.

I tried removing the columns from the controller (as they aren't dynamic, but the error is still here)

Pluto grid initialization:

```
PlutoGrid(
  columns: colunasNota,
  rows: _addnotaController.rows,
  onLoaded: (PlutoGridOnLoadedEvent event) {
    _addnotaController.stateManager = event.stateManager;
    _addnotaController.stateManager
        ?.setShowColumnFilter(true);
    _addnotaController.stateManager
        ?.setSelectingMode(PlutoGridSelectingMode.none);
  },
  configuration: defaultPluto,
  createFooter: (stateManager) {
    stateManager.setPageSize(100, notify: false);
    return PlutoPagination(stateManager);
  },
  mode: PlutoGridMode.select,
),
```

Complete sample code:

[search.zip](https://github.com/bosskmk/pluto_grid/files/8043448/search.zip)

Error code:

```
════════ Exception caught by scheduler library ═════════════════════════════════
The following StateError was thrown during a scheduler callback:
Bad state: No element

When the exception was thrown, this was the stack
[C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart]() 251:49  throw_
[C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/core/iterable.dart]() 624:7                                         get first
packages/pluto_grid/src/manager/state/column_state.dart 219:46                                                             get currentColumn
packages/pluto_grid/src/manager/state/editing_state.dart 54:56                                                             get autoEditing
packages/pluto_grid/src/manager/state/cell_state.dart 212:16                                                               setCurrentCell
packages/pluto_grid/src/pluto_grid.dart 386:23                                                                             <fn>
packages/flutter/src/scheduler/binding.dart 1144:15                                                                        [_invokeFrameCallback]
packages/flutter/src/scheduler/binding.dart 1089:9                                                                         handleDrawFrame
packages/flutter/src/scheduler/binding.dart 862:7                                                                          <fn>
[C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/isolate_helper.dart]() 48:19       internalCallback

```

**Comment by Macacoazul01** [comment]
@bosskmk any updates?

**Comment by Macacoazul01** [comment]
the problem happens on this function of ./src/pluto_grid.dart':


```
void _initSelectMode() {
    if (widget.mode.isSelect != true) {
      return;
    }

    WidgetsBinding.instance!.addPostFrameCallback((_) {
      if (_stateManager.currentCell == null && widget.rows.isNotEmpty) {
        _stateManager.setCurrentCell(
         widget.rows.first.cells.entries.first.value, 0);
      }

      _stateManager.gridFocusNode!.requestFocus();
    });
  }
```

commenting lines 386 and 387 solved my problem (`_stateManager.setCurrentCell(widget.rows.first.cells.entries.first.value, 0);`)

**Comment by Macacoazul01** [comment]
on 2.10.0 the lines are 352-355

**Comment by bosskmk** [comment]
@Macacoazul01 
The problem was corrected and distributed in version 2.10.1.
Please let me know if the issue occurs again with that version.
thank you.
